pkg/controlsvc/controlsvc.go

```diff
@@ -115,6 +116,10 @@ func (s *Server) AddControlFunc(name string, cType ControlCommandType) error {
 func (s *Server) RunControlSession(conn net.Conn) {
 	logger.Info("Client connected to control service\n")
 	defer func() {
+		if conn == nil {
+			logger.Debug("nil context")
```

Comment:
I think `conn` stands for connection. Maybe something like "Connection already closed, continuing."

Reply:
I have changed it , I think that the problem lies somewhere else.

Reply:
I think this is only failing because we have already panicked elsewhere.  Note that `conn` is a local variable which is never assigned in the body of this function, so it can't become `nil` as a result of local code, and it would fail on the first write if it was passed in as a nil.